bindings/c/include/vw/experimental/workspace.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#pragma once
+
+#include "vw/experimental/types.h"
+#include "vw/experimental/vwvis.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+  VW_DLL_PUBLIC VWStatus vw_create_workspace(VWOptions* options_handle, VWTraceMessageFunc* trace_listener,
+      void* trace_context, VWWorkspace** output_handle, VWErrorString* err_str_container) VW_API_NOEXCEPT;
```

Comment:
`VWErrorString*` do we want to turn this into a more fully featured `VWErrorInfo` object which has been discussed including things such as a call stack?

Reply:
Even if `VWErrorInfo` only initially contains an error string I think it offers more flexibility as the error code design evolves